lib/sycamore/sycamore/transforms/sort.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any, Optional
+
+from sycamore.plan_nodes import Node, Transform
+from sycamore.data import Document
+
+from ray.data import Dataset
+
+
+class Sort(Transform):
+    """
+    Sort by field in Document
+    """
+
+    def __init__(self, child: Node, descending: bool, field: str, default_val: Optional[Any] = None):
+        super().__init__(child)
+        self._descending = descending
+        self._field = field
+        self._default_val = default_val
+
+    def execute(self, **kwargs) -> "Dataset":
+        # creates dataset
+        ds = self.child().execute(**kwargs)
+
+        # adds a "key" column containing desired field
+        map_fn = self.make_map_fn_sort()
+        ds = ds.map(map_fn)
+
+        # sorts the dataset
+        ds = ds.sort("key", descending=self._descending)
+        ds = ds.drop_columns(["key"])
+        return ds
+
+    def make_map_fn_sort(self):
+        def ray_callable(input_dict: dict[str, Any]) -> dict[str, Any]:
+            doc = Document.from_row(input_dict)
+
+            try:
+                fields = self._field.split(".")
+                val = getattr(doc, fields[0])
+                if len(fields) > 1:
+                    for f in fields[1:]:
```

Comment:
Is the idea that index 0 is "properties" so you skip it? Might be worth asserting that just in case someone passes in something weird.